### PR TITLE
refactor!: Remove tracking of "culprit" from Require methods

### PR DESCRIPTION
**Pull request by niloc132**
The string `culprit` as a marker for pointing to an issue in a stack trace has never appeared in a DHC issue, nor in DHC slack, and the last DHE slack message to use it in a mildly helpful way was more than two years ago.

This patch removes the functionality from `RequirementFailure`, and also removes any overload of a method in `Require` that would allow callers to customize the depth within the stack that should be marked.

Partial #188

### CODE REVIEW

**Comment by niloc132** [comment]
The first commit simplifies RequirementFailure to match AssertionFailure, and can stand alone even if we want to drop the second.

I checked Core+, and RequirementFailure is never explicitly used, nor are any of the overloads that take a depth param.

The patch presently increases the size of the file rather than shrinking it - I think I accidentally inlined something that was used in many places, probably one of the `fail()` overloads. I'll take a quick attempt at undoing that and rewrite the second commit.

File: Base/src/test/java/io/deephaven/base/TestLowGarbageArrayIntegerMap.java [hunk 1]
```diff
@@ -60,7 +60,6 @@ public void testLowGarbageArrayIntegerMap() {
             integerToStringMap.put(-1, "negative one");
             fail("expected bad index to fail");
         } catch (RequirementFailure requirementFailure) {
```
Comment (rcaudy) [inline Base/src/test/java/io/deephaven/base/TestLowGarbageArrayIntegerMap.java:62]:
```suggestion
        } catch (RequirementFailure expected) {
```

**Review by rcaudy — COMMENTED** [review]
I visually scanned all of the refactoring changes, looks clean. Please run nightlies, just in case.

**Review by rcaudy — APPROVED** [review]

**Comment by niloc132** [comment]
Nightly build passed: https://github.com/niloc132/deephaven-core/actions/runs/9897209491

**Pull request merged**